src/ucp/core/ucp_request.h

```diff
@@ -447,7 +447,7 @@ struct ucp_request {
             ucp_send_nbx_callback_t cb;         /* Completion callback */
             uct_worker_cb_id_t      prog_id;    /* Progress callback ID */
             int                     comp_count; /* Countdown to request completion */
-            ucp_ep_ext_gen_t        *next_ep;   /* Next endpoint to flush */
+            ucp_ep_ext_t            *next_ep;   /* Next endpoint to flush */
```

Comment:
maybe point to ep

Reply:
We can't do it because of how flushing flow works.

Currently, in `ucp_worker_flush_req_set_next_ep` it might set `worker->all_eps` as the "next EP ext" to flush, then [this](https://github.com/openucx/ucx/blob/master/src/ucp/rma/flush.c#L491) ot [this](https://github.com/openucx/ucx/blob/master/src/ucp/rma/flush.c#L461) condition will be false and `set_next_ep` is skipped which is OK.

If we store EP instead of EP ext, need to [dereference](https://github.com/openucx/ucx/blob/master/src/ucp/rma/flush.c#L419) from an invalid `ep_ext` back to `ep`, which can't work, because then need to deref back from that `ep` to `ep_ext`. Then, the above-mentioned `if` conditions won't check correctly -> `set_next_ep` will be reached pointing to invalid memory as "EP".